src/sonic-config-engine/minigraph.py

```diff
@@ -1666,6 +1666,15 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
         if port_name in mgmt_alias_reverse_mapping.keys():
             continue
 
+        port_default_speed =  port_speeds_default.get(port_name, None)
+        port_png_speed = port_speed_png[port_name]
+
+        # when the port speed is changes from 400g to 100g 
+        # update the port lanes, use the first 4 lanes of the 400G port to support 100G port
+        if port_default_speed == '400000' and port_png_speed == '100000':
+            port_lanes =  ",".join(ports[port_name].get('lanes', '').split(',')[:4])
+            ports[port_name]['lanes'] = port_lanes
```

Comment:
Just a question -- Is this method of taking the first four Serdes lanes when moving from 400G --> 100G is ok to be applied to non-voq and pizzabox devices also ?

Reply:
we need to validate this condition for our platforms and will revert back. Shouldn't be platform specific?

Reply:
added a check for` switch_type = 'voq'` in the latest commit

Reply:
Do we need a check for voq only? Most of non voq platforms support the first 4 lanes for 100G speed. For the rest, we need to check if modifying lane ordering in port_config.ini will work. If not, we will have to add additional support. If we do not have this check for voq, this won't be breaking any functionality for supported platforms. So I think we should be good without this check.

Reply:
@anamehra, we are currently testing only voq linecards. so added the check to limit the scope of the change in this PR.
When we validate this feature on other platform when can relax the check

Reply:
@abdosi , do you know what are the other non voq platforms, other than chassis-packet, could be affected by this? I think this check for voq could be dropped. Any other way for a platform to decide whether to use or not this feature by checking compatibility?
Also, is it possible that the platform could provde the lane information to be used for 400G/100G instead of assuming the first 4 lanes?

Reply:
@anamehra agree once we have use case we can remove voq check as needed. To have backward compatibility it is better to have voq check for now.  

Regarding platform change i think that is another option for this. We can think on this further in subsequent enhancement